sentry_streams/tests/pipeline/test_msg_parser.py

```diff
@@ -24,3 +28,31 @@ def test_msg_serializer_custom_dt_format() -> None:
     result_bytes = msg_serializer(mock_msg, dt_format=dt_format)
     result_str = result_bytes.decode("utf-8")
     assert json.loads(result_str) == {"timestamp": dt.strftime(dt_format)}
+
+
+class FakeMessage:
+    def __init__(self, payload: Sequence[bytes], schema: Any) -> None:
+        self.payload = payload
+        self.schema = schema
+
+
+def test_batch_msg_parser_nominal_case(monkeypatch: MonkeyPatch) -> None:
+    input_payload = [b'{"a": 1}', b'{"b": 2}']
+
+    expected_output = [{"a": 1}, {"b": 2}]
+
+    class FakeCodec:
+        def decode(self, payload: bytes, _: Any) -> Mapping[str, int]:
+            if payload == b'{"a": 1}':
+                return {"a": 1}
+            else:
+                return {"b": 2}
```

Comment:
instead of returning a fake codec here i would suggest to use Codec class by kafka schemas directly, just to exercise more code. if there is a breaking change in sentry-kafka-schemas (intentional or not), we're more likely to catch it with this test

same for FakeMessage, it seems feasible to me to create a real message?